src/platform/Ameba/ConfigurationManagerImpl.cpp

```diff
@@ -152,6 +152,28 @@ CHIP_ERROR ConfigurationManagerImpl::GetPrimaryWiFiMACAddress(uint8_t * buf)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ConfigurationManagerImpl::GetUniqueId(char * buf, size_t bufSize)
+{
+#ifdef CHIP_DEVICE_CONFIG_UNIQUE_ID
+    ReturnErrorCodeIf(bufSize < sizeof(CHIP_DEVICE_CONFIG_UNIQUE_ID), CHIP_ERROR_BUFFER_TOO_SMALL);
+    strcpy(buf, CHIP_DEVICE_CONFIG_UNIQUE_ID);
+    return CHIP_NO_ERROR;
+#else
+    char temp[32];
+    int i = 0, j = 0;
+
+    wifi_get_mac_address(temp);
```

Comment:
what is the size of wifi_get_mac_address?

The loop below loops by bufSize, but reads data from temp. So two issues:
  - a large bufSize will cause uninitialized memory to be read and can  go beyond temp
  - even if bufSize is <= 32, if wifi_get_mac_address does not fill temp (and (2 hex chars  + 1 separator) * 6 is only 18) this means that the read can still go outside temp initialized data read.

Reply:
The wifi_get_mac_address returns 0 if get mac address is successful and -1 for error.
This implementation assumes kMaxUniqueIDLength is fixed to be 32.
To address the issues you pointed, I have added error check for 'bufSize !=32' and 'wifi_get_mac_address is not successful'.
Thanks!